# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsFocusManager.h
# Commit: 2425c93f20f6
# Full Hash: 2425c93f20f61b806df955141332d7cb553c6efb
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2024-01-17 04:08:25
# Regressor Bug: 779684
# File Overlap Count: 1
# Description:
#   Bug 1872301 - Make `nsFocusManager::GetSelectionLocation` never fail when found text node has no frame r=NeilDeakin
#   
#   In the test case, `nsFocusManager::GetSelectionLocation` is called with
#   collapsed selection at end of the `<svg>` which ends with collapsible
#   white-spaces.  Therefore, it reaches the white-space only text node and it
# ==============================================================================

diff -r ad6d17afc267 -r 2425c93f20f6 dom/base/nsFocusManager.h
--- a/dom/base/nsFocusManager.h	Wed Jan 17 00:18:05 2024 +0000
+++ b/dom/base/nsFocusManager.h	Wed Jan 17 00:25:04 2024 +0000
@@ -548,10 +548,9 @@
    * Retrieves the start and end points of the current selection for
    * aDocument and stores them in aStartContent and aEndContent.
    */
-  nsresult GetSelectionLocation(Document* aDocument,
-                                mozilla::PresShell* aPresShell,
-                                nsIContent** aStartContent,
-                                nsIContent** aEndContent);
+  void GetSelectionLocation(Document* aDocument, mozilla::PresShell* aPresShell,
+                            nsIContent** aStartContent,
+                            nsIContent** aEndContent);
 
   /**
    * Retrieve the next tabbable element in scope owned by aOwner, using